cmd/frontend/service.go

```diff
@@ -428,12 +428,34 @@ func (fes *FrontEndService) WriteConfigGW(conf *string) {
 				nbr := strings.Split(gw.Address, "/")[0]
 				*conf += "protocol bgp 'NBR-" + gw.Name + "' from LINK {\n"
 				*conf += "\tinterface \"" + fes.extInterface + "\";\n"
-				// currently Gateway does not hold much BGP specific details, using common FE values
-				*conf += "\tlocal port " + fes.localPortBGP + " as " + fes.localAS + ";\n"
-				*conf += "\tneighbor " + nbr + " port " + fes.remotePortBGP + " as " + fes.remoteAS + ";\n"
+
+				// session specific BGP params
+				localASN := fes.localASN
+				localPort := fes.localPortBGP
+				remoteASN := fes.remoteASN
+				remotePort := fes.remotePortBGP
+				if gw.LocalASN != 0 {
+					localASN = strconv.FormatUint(uint64(gw.LocalASN), 10)
+				}
+				if gw.LocalPort != 0 {
+					localPort = strconv.FormatUint(uint64(gw.LocalPort), 10)
+				}
+				if gw.RemoteASN != 0 {
+					remoteASN = strconv.FormatUint(uint64(gw.RemoteASN), 10)
+				}
+				if gw.RemotePort != 0 {
+					remotePort = strconv.FormatUint(uint64(gw.RemotePort), 10)
+				}
```

Comment:
CRD is using uint32 for ASNs and uint16 for ports. Maybe better to keep them consistent?